### ISSUE TITLE
Contradictory error message when calling function with type arguments inconsistent with arguments

### ISSUE THREAD

**Issue by mattmccutchen** [Microsoft/TypeScript#26835]
Found via https://stackoverflow.com/questions/52130789 .

**TypeScript Version:**  master (4510149)

**Search Terms:** "expected" "arguments but got" type parameter argument

**Code**
```ts
declare function foo<T>(a: T, b: T);
declare function foo(a: {});
foo<string>("hello");
```

**Expected behavior:** An error message that makes sense.  Perhaps `Expected 2 arguments, but got 1` based on the number of type arguments passed.

**Actual behavior:** Error: `Expected 1-2 arguments, but got 1.`

**Playground Link:** [link](https://www.typescriptlang.org/play/#src=declare%20function%20foo%3CT%3E(a%3A%20T%2C%20b%3A%20T)%3B%0D%0Adeclare%20function%20foo(a%3A%20%7B%7D)%3B%0D%0Afoo%3Cstring%3E(%22hello%22)%3B)

**Related Issues:** didn't find any
